tests/ratingservice/main_test.py

```diff
@@ -63,35 +73,36 @@ def testNewRatingCalculation(self):
         # use products[1] to avoid counting new vote from testRate() test
         test_product_id = products[1]
         new_rating_vote = 5
-        url_get = self.composeUrl("rating", test_product_id)
-        url_post = self.composeUrl("rating")
-        url_put = self.composeUrl("recollect")
+        url_get = composeUrl("rating", test_product_id)
+        url_post = composeUrl("rating")
+        url_put = composeUrl("recollect")
 
         # get current rating / post new vote / recollect / get updated rating
-        result1 = requests.get(url_get)
+        result1 = self.session.get(url_get, timeout=1.0)
         self.assertEqual(result1.status_code, 200)
-        result2 = requests.post(url_post, json={
+        result2 = self.session.post(url_post, timeout=1.0, json={
             'rating': new_rating_vote,
             'id': test_product_id
         })
         self.assertEqual(result2.status_code, 200)
-        result2 = requests.put(url_put)
+        result2 = self.session.put(url_put, timeout=1.0)
         self.assertEqual(result2.status_code, 200)
-        result2 = requests.get(url_get)
+        result2 = self.session.get(url_get, timeout=1.0)
         self.assertEqual(result2.status_code, 200)
 
-        result1_data = result1.json()
-        result2_data = result2.json()
-        self.assertEqual(result2_data['votes'],
-                         result1_data['votes'] + 1)
-
-        # new rating should be: old_rating+((5.0-old_rating)/(old_num_of_votes+1))
-        new_rating = result1_data['rating'] + \
-            ((new_rating_vote - result1_data['rating']) /
-             (result1_data['votes'] + 1))
-        # compare new rating vs. expected which was rounded to 4 digits after floating point
-        self.assertEqual(result2_data['rating'],
-                         math.ceil(new_rating*10000)/10000)
+        data = result1.json()
+        prev_vote = data['votes']
+        prev_rating = decimal.Decimal(data['rating'])
+        data = result2.json()
+        new_vote = data['votes']
+        new_rating = decimal.Decimal(data['rating'])
+        self.assertEqual(new_vote, prev_vote + 1)
+        expected_rating = prev_rating + \
+            ((new_rating_vote-prev_rating)/(prev_vote+1))
+        # compare expected result rounded to 4 decimal places
+        QUANTIZE_VALUE = decimal.Decimal("0.0001")
+        self.assertEqual(new_rating.quantize(QUANTIZE_VALUE),
```

Comment:
Is there a reason you're quantizing instead of setting the decimal precision to 4?

What does `data['rating']`look like? Will it be 4 digits?

Reply:
Yes, it is a floating point value with maximum 4 digits after the point.

Reply:
According to the [documentation](https://docs.python.org/3/library/decimal.html):

> Context precision and rounding only come into play during arithmetic operations

The main use is to keep precision when converting from float.